ddtrace/tracer/option.go

```diff
@@ -181,9 +181,15 @@ func forEachStringTag(str string, fn func(key string, val string)) {
 func newConfig(opts ...StartOption) *config {
 	c := new(config)
 	c.sampler = NewAllSampler()
-	c.agentAddr = resolveAgentAddr()
+	c.agentURL = "http://" + resolveAgentAddr()
```

Comment:
Should we not be defaulting to HTTPS? If there is a particular reason why we shouldn't, we should have a comment here explaining why we are defaulting to the less secure option.

Reply:
I see this conversation was resolved, what was the end result? Was it determined http is fine? Or is it overwritten later and this is just the default? Does this need a comment or something to clarify for future readers?

Reply:
@ajgajg1134 thanks for the review! 

> I see this conversation was resolved, what was the end result?


Sorry, this was resolved via Slack discussion and I didn't add a comment here. We were already using http before this PR ([here](https://github.com/DataDog/dd-trace-go/blob/v1.43.1/ddtrace/tracer/transport.go#L104)). I don't believe the agent supports https requests for traces, the need to support https in `DD_TRACE_AGENT_URL` is for edge cases where the user has a layer in front of the agent, such as a SSL-terminating load balancer or a proxy. Let me know if you think I should add this as a comment ? :)